### PR TITLE
Ci workflow

### PR DESCRIPTION
**Pull request by LeNelSoN**

### CODE REVIEW

File: .github/workflows/test-and-build.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - published
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 17
```
Comment (padabou) [inline .github/workflows/test-and-build.yml:21]: you should step up to 21

File: .github/workflows/test-and-build.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'zulu'
```
Comment (padabou) [inline .github/workflows/test-and-build.yml:25]: if possible move to temurin ;-)

File: .github/workflows/test-and-build.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+      - name: Build project
+        run: mvn clean install
```
Comment (padabou) [inline .github/workflows/test-and-build.yml:44]: why split test and install ? install made the test if you do not add skipTests

File: .github/workflows/test-and-build.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
+
+      - name: Build Docker image
+        run: |
+          # Build image with the version of the release
+          docker build -t ghcr.io/LeNelSoN/you-are-the-hero:${{ github.event.release.tag_name }} .
+          # Tag image as latest
+          docker tag ghcr.io/LeNelSoN/you-are-the-hero:${{ github.event.release.tag_name }} ghcr.io/your-github-username/you-are-the-hero:latest
+
+      - name: Push Docker images
```
Comment (padabou) [inline .github/workflows/test-and-build.yml:64]: nice !

**Review by padabou — APPROVED** [review]

File: .github/workflows/test-and-build.yml (hunk 1 above)
Comment (LeNelSoN) [reply .github/workflows/test-and-build.yml:21]: For now, I will keep JDK 17 to maintain consistency across the project. I plan to consider an upgrade to JDK 21 in the future.

File: .github/workflows/test-and-build.yml (hunk 3 above)
Comment (LeNelSoN) [reply .github/workflows/test-and-build.yml:44]: The purpose of separating the test and build steps is to reduce the execution time of the pipeline, especially if the tests fail. This allows us to quickly identify any issues without waiting for the entire build phase to complete.

You're correct that it is also possible to skip the tests during the build phase, which can further optimize the process.

**Pull request merged**